src/goto-synthesizer/cegis_verifier.cpp

```diff
@@ -666,6 +668,7 @@ optionalt<cext> cegis_verifiert::verify()
     // assignable violation found
     if(property.second.description.find("assignable") != std::string::npos)
     {
+      target_violation_found = true;
```

Comment:
Perhaps you should use `optionalt<struct to be defined>` here? You have several `target_*` variables here that need to be kept in sync. How about putting (what currently is) `target_violation` and `target_violation_info` in a struct type and then declare an `optionalt` of that type before the loop? Your `target_violation_found` will then be `.has_value()` of that `optionalt`.

Reply:
I use ```propertiest::iterator``` instead, which already contain the id and the information of the violated property. When target violation is not found, the iterator will equal to the end of ```properties```.